### PR TITLE
 Allow streams of non-english speaking streamers #41 

### PR DESCRIPTION
**Pull request by felixdorn**
image

I've added a flag in the homepage next to the title.
I've also added some tests for this feature.
And last but not least i've required [`calebporzio/sushi`](github.com/calebporzio/sushi) so there is an Eloquent relation between `Stream` and `Language` which could allow later on more complex filters if it's ever needed. And it's also super clean.

### CODE REVIEW

**Comment by felixdorn** [comment]
> A flag represents a nation, not a language. This is an extremely widespread wrong practice.

I completly agree, I think it would be better to just show "English", "French" based on the poll's results. That would simplify the code quite a lot as well so it's a win-win.

**Comment by christophrumpel** [comment]
Yeah, I think so too. So we might only need the name "English" and key "EN". So we can decide depending on the design which one to use.

**Comment by felixdorn** [comment]
I've removed all the flag stuff added yesterday. For the record, I don't think it's the best spot nor the best way to show the stream's language. But I'm curious about what @Larsklopstra thinks?

image

File: database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddLanguageCodeToStreamsTable extends Migration
+{
+    public function up()
+    {
+        Schema::table('streams', function (Blueprint $table) {
+            $table->string('language_code')->default('en');
```
Comment (christophrumpel) [inline database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php:12]: Please add the new field after `status`.

File: resources/views/pages/partials/stream.blade.php [hunk 2]
```diff
@@ -43,9 +47,9 @@ class="flex px-2 py-1 space-x-2 bg-gray-900 rounded bg-opacity-80 backdrop-filte
 
         <ul class="flex flex-wrap gap-6">
             <li>
-                <a href="{{ $stream->toWebcalLink() }}"
+                <a href="{{ route('calendar.ics.stream', $stream) }}"
```
Comment (christophrumpel) [inline resources/views/pages/partials/stream.blade.php:50]: I think this change was from an old stage. Might be it came in through a conflict? The `toWebcalLink` method should be used.

File: resources/views/livewire/import-youtube-live-stream.blade.php [hunk 3]
```diff
@@ -27,7 +27,15 @@
                                     <input type="text" wire:model.defer="youtubeId" id="youtubeStreamId"
                                            class="mt-1 focus:ring-indigo-500 focus:border-indigo-500 block w-full shadow-sm sm:text-sm border-gray-300 rounded-md">
                                 </div>
-
+                                <div class="col-span-6 sm:col-span-3">
```
Comment (christophrumpel) [inline resources/views/livewire/import-youtube-live-stream.blade.php:30]: Please add the language `behind` the date for now. We can still work on design improvements later if needed.

**Review by christophrumpel — CHANGES_REQUESTED** [review]

**Comment by Larsklopstra** [comment]
@felixdorn I like that position. Would only make the color white and hide the language if it is English

**Comment by Larsklopstra** [comment]
Could you also change the stroke width to match the calendar?

File: resources/views/pages/partials/stream.blade.php (hunk 2 above)
Comment (felixdorn) [reply resources/views/pages/partials/stream.blade.php:50]: Fixed!

File: resources/views/livewire/import-youtube-live-stream.blade.php (hunk 3 above)
Comment (felixdorn) [reply resources/views/livewire/import-youtube-live-stream.blade.php:30]: Fixed!

File: database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php (hunk 1 above)
Comment (felixdorn) [reply database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php:12]: Fixed!

**Comment by felixdorn** [comment]
Language on the left side : 
image

Language on the right side : 
image

I agree with Lars, it looks better on the left side, what do you think ?

**Comment by christophrumpel** [comment]
Thanks, guys. I only want to make sure the date is super easy to read. This means for me that all dates should be at the same place (beneath each other). So we can:

* Always show the language on the left (including English)
* Or place the lang on the right or somewhere else

**Comment by SamuelNitsche** [comment]
What about something like this? I think on the right there is plenty of space which can be used.

Bildschirmfoto 2021-05-21 um 15 03 46

File: database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php (hunk 1 above)
Comment (christophrumpel) [reply database/migrations/2021_05_20_200000_add_language_Code_to_streams_table.php:12]: I believe it was changed again by the merge?

**Review by christophrumpel — CHANGES_REQUESTED** [review]

**Comment by christophrumpel** [comment]
@SamuelNitsche yeah this would also work for me. What about you @Larsklopstra?

**Comment by Larsklopstra** [comment]
I prefer @felixdorn's with the language on the right

**Comment by christophrumpel** [comment]
Alright, thanks. Then please @felixdorn put it on the right and we can merge. When there are more streams on the list in the future (including in  non-English) we can check again if it works 👍

File: resources/views/pages/partials/stream.blade.php [hunk 4]
```diff
@@ -18,18 +18,23 @@ class="flex px-2 py-1 space-x-2 bg-gray-900 rounded bg-opacity-80 backdrop-filte
...
+                 src="{{ $stream->thumbnail_url }}"
+                 alt="Video thumbnail"/>
         </figure>
     </a>
 
     <article class="flex flex-col items-start p-6 space-y-2 bg-gray-600 rounded-b-xl lg:rounded-xl">
-        <x-local-time class="inline-block font-bold tracking-tight text-red-400"
-            :date="$stream->scheduled_start_time" />
+        <div class="flex items-center">
+            @if ($stream->language->shouldRender())
```
Comment (christophrumpel) [inline resources/views/pages/partials/stream.blade.php:28]: please put the lang on the right.

**Review by christophrumpel — APPROVED** [review]

**Comment by christophrumpel** [comment]
Thanks @felixdorn!

**Pull request merged**